tests/components/squeezebox/test_config_flow.py

```diff
@@ -166,6 +166,33 @@ async def patch_async_query(self, *args):
     assert result["errors"] == {"base": "invalid_auth"}
 
 
+async def test_form_validate_exception(hass: HomeAssistant) -> None:
+    """Test we handle exception."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "edit"}
+    )
+
+    class FakeError(exceptions.HomeAssistantError):
+        """Error."""
+
+    async def patch_async_query(self, *args):
+        raise FakeError
+
+    with patch("pysqueezebox.Server.async_query", new=patch_async_query):
```

Comment:
```suggestion
    with patch("homeassistant.components.squeezebox.config_flow.Server.async_query", side_effect=Exception):
```

Reply:
done

Reply:
actually, no, that doesn't work.  This is a validation on a step_id = "edit", so isn't the context correct as "edit".  SOURCE_USER aka "user" gives a schema error.

Reply:
But that is also what I mean, our config flow tests should be end to end, so you start with either a discovery method or a user step. You don't start with the flow halfway

Reply:
So, you want the invalid tests to start with a valid user step - rather than just test the exception.  I guess I get it, and understand it's a better approach.  There's gonna be a lot of tests failing that measure I guess ;-)

Reply:
Haha, I know, and a good way to make sure that happens is by not allowing new ones to be added :D

Reply:
For me, and I imagine many maintainers, writing testing is a new world.  We're just doing this as a hobby to try and contribute something back, but I've never written testing code before so everything about this is new for me.

Reply:
But don't forget, if you have questions you can always ask them :D

Reply:
I just posted an expanded version - grateful for comments and thoughts....